controllers/contractController.js

```diff
@@ -0,0 +1,39 @@
+var Web3 = require('web3');
+var contract = require('../models/contract.js');
+var contractFunction = require('../models/contractFunction.js');
+var sqsHelper = require('../helpers/aws/sqsHelper.js');
+var web3 = new Web3();
+
+web3.setProvider(new web3.providers.HttpProvider(process.env.ENODE_BASE || 'http://localhost:8545'));
+
+function ContractController() {}
+
+ContractController.prototype.setContractFunctionData = function(contractId, contractFunctionId, data) {
+	return contract.read(contractId).then(function(contractResult) {
+		if (contractResult.rowCount > 0) {
+			var contractABI = JSON.parse(contractResult.rows[0].abi);
```

Comment:
JSON.parse interprets anything you send to it as a string. It does the equivalent of String(input), meaning literal true, false, 25, null will be accepted. undefined won't, because that isn't valid JSON per section 15.12.1.2 in the spec you posted in the other comment.

#ref http://stackoverflow.com/questions/17648150/how-does-json-parse-manage-undefined